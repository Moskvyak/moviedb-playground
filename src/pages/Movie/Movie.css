.movie__backdrop-wrapper {
    width: 100%;
}
.movie__backdrop-wrapper img {
    width: 100%;
}
.movie__title-wrapper {
    color: #fff;
    margin-left: 155px;
    margin-right: 20px;
    margin-top: 20px;
    padding-bottom: 20px;
    font-size: 1.25em;
    font-weight: 600;
    text-align: left;
}
.movie__title-section {
    position: relative;
    padding-bottom: 20px;
}
.movie__back-button {
    position: absolute;
    left: 20px;
    top: 20px;
    width: 50px;
    height: 50px;
    z-index: 1;
}
.movie__poster-wrapper {
    position: absolute;
    bottom: 20px;
    left: 20px;
    width: 120px;
}

.movie__poster-wrapper img {
    width: 100%;
    border-radius: 20px;
}

.movie__extra-info {
    font-size: 0.66em;
    margin-top: 10px;
    margin-bottom: 10px;
    line-height: 1.25em;
}
.movie__section {
    font-size: 0.9em;
    padding: 0 20px 40px 20px;
}
.movie__section h3 {
    margin: 10px 0;
    padding-top: 20px;
    border-top: 1px solid lightslategray;
    font-size: 1.5em;
}

@media screen and (min-width: 960px) {
    .movie__header {
        float: left;
        width: 45%;
    }
    .movie__section {
        float: right;
        width: 45%;
        font-size: 1.25em;
    }
    .movie__section h3 {
        border: none;
    }
}