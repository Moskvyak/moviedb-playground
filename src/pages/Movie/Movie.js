import React from 'react';
import { getMovie } from '../../services/Dataservice';
import { withRouter } from 'react-router-dom';
import './Movie.css';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';
export class Movie extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      movie: {}
    };
  }
  goBack = () => {
    this.props.history.goBack();
  };
  componentDidMount = async () => {
    const { match } = this.props;
    if (match && match.params && match.params.id) {
      const movie = await getMovie(match.params.id);
      this.setState({ movie });
    }
  };
  render() {
    const {
      backdrop_path,
      poster_path,
      title,
      vote_average,
      release_date,
      runtime,
      overview
    } = this.state.movie;
    const backDropFileSize = '/w500';
    const posterFileSize = '/w200';
    const rating = `${vote_average * 10} %`;
    const backDropUrl = `${process.env
      .REACT_APP_ASSETS_URL}${backDropFileSize}${backdrop_path}`;
    const posterUrl = `${process.env
      .REACT_APP_ASSETS_URL}${posterFileSize}${poster_path}`;

    const date = new Date(release_date).toLocaleString('en-us', {
      year: 'numeric'
    });
    const movieHours = parseInt(runtime / 60);
    const movieLength = `${movieHours}h ${runtime - movieHours * 60}min`;

    return (
      <div className="page-background-color page-wrapper primary-color movie__page">
        <div className="movie__header">
          <div className="movie__backdrop-wrapper">
            <div className="movie__back-button" onClick={this.goBack}>
            <FontAwesomeIcon
              className="searchform__right-icon"
              icon={faArrowLeft}
              color={'#FFF'}
              size="1x"
              
              />
              </div>
            <img src={backDropUrl} />
          </div>
          <div className="movie__title-section">
            <div className="movie__poster-wrapper">
              <img src={posterUrl} />
            </div>
            <div className="movie__title-wrapper">
              {title}
              <div className="movie__extra-info color-lightgray">
                {date} &#183; {rating} User Score
                <br />
                {movieLength}
              </div>
            </div>
          </div>
        </div>
        <div className="movie__section">
          <h3 className="color-white"> Overview</h3>
          <div className="color-lightgray">
            {overview}
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(Movie);
