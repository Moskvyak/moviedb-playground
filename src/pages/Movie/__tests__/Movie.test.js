import React from 'react';
import renderer from 'react-test-renderer';
import { Movie } from '../Movie';

describe('<Movie />', () => {
  it('should render with required props', () => {
    const component = renderer.create(<Movie />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
