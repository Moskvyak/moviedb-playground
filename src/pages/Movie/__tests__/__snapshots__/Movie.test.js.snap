// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movie /> should render with required props 1`] = `
<div
  className="page-background-color page-wrapper primary-color movie__page"
>
  <div
    className="movie__header"
  >
    <div
      className="movie__backdrop-wrapper"
    >
      <div
        className="movie__back-button"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-arrow-left fa-w-14 fa-1x searchform__right-icon"
          color="#FFF"
          data-icon="arrow-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <img
        src="undefined/w500undefined"
      />
    </div>
    <div
      className="movie__title-section"
    >
      <div
        className="movie__poster-wrapper"
      >
        <img
          src="undefined/w200undefined"
        />
      </div>
      <div
        className="movie__title-wrapper"
      >
        <div
          className="movie__extra-info color-lightgray"
        >
          Invalid Date
           Â· 
          NaN %
           User Score
          <br />
          NaNh NaNmin
        </div>
      </div>
    </div>
  </div>
  <div
    className="movie__section"
  >
    <h3
      className="color-white"
    >
       Overview
    </h3>
    <div
      className="color-lightgray"
    />
  </div>
</div>
`;
